src/dryrun.ts

```diff
@@ -167,3 +167,159 @@ export async function createDryrun({
+    return this.trace.map((dtl) => dtl.traceLine());
+  }
+}
+class DryrunTransactionResult {
+  defaultSpaces: number = 50;
+
+  disassembly: string[] = [];
+  appCallMessages: string[] | undefined = [];
+  localDeltas: any[] | undefined = [];
+  globalDelta: any[] | undefined = [];
+  cost: number | undefined = 0;
+  logicSigMessages: string[] | undefined = [];
+  logicSigDisassemly: string[] | undefined = [];
+  logs: string[] | undefined = [];
+
+  appCallTrace: DryrunTrace | undefined = undefined;
+  logicSigTrace: DryrunTrace | undefined = undefined;
+
+  required = ['disassembly'];
+  optionals = [
+    'app-call-messages',
+    'local-deltas',
+    'global-delta',
+    'cost',
+    'logic-sig-messages',
+    'logic-sig-disassembly',
+    'logs',
+  ];
+
+  traces = ['app-call-trace', 'logic-sig-trace'];
+
+  constructor(dtr: Record<string, any>) {
+    this.disassembly = dtr.disassembly;
+    this.appCallMessages = dtr['app-call-messages'];
+    this.localDeltas = dtr['local-deltas'];
+    this.globalDelta = dtr['global-delta'];
+    this.cost = dtr.cost;
+    this.logicSigMessages = dtr['logic-sig-messages'];
+    this.logicSigDisassemly = dtr['logic-sig-disassembly'];
+    this.logs = dtr.logs;
+    this.appCallTrace = new DryrunTrace(dtr['app-call-trace']);
+    this.logicSigTrace = new DryrunTrace(dtr['logic-sig-trace']);
+  }
+
+  appCallRejected(): boolean {
+    return (
+      this.appCallMessages !== undefined &&
+      this.appCallMessages.includes('REJECT')
+    );
+  }
+
+  logicSigRejected(): boolean {
+    return (
+      this.logicSigMessages !== undefined &&
+      this.logicSigMessages.includes('REJECT')
+    );
+  }
+
+  trace(drt: DryrunTrace, disassembly: string[], spaces?: number): string {
+    // eslint-disable-next-line no-param-reassign
+    if (spaces === undefined) spaces = this.defaultSpaces;
+
+    const lines = [`pc# line# source${' '.repeat(spaces - 16)}stack`];
+    for (const [line, pc, stack] of drt.getTrace()) {
+      const linePadding = ' '.repeat(4 - line.toString().length);
+      const pcPadding = ' '.repeat(4 - pc.toString().length);
+      const dis = disassembly[line];
+
+      const srcLine = `${pcPadding}${pc} ${linePadding}${line} ${dis}`;
+
+      const stackPadding = ' '.repeat(Math.max(1, spaces - srcLine.length));
+
+      lines.push(`${srcLine}${stackPadding}${stack}`);
+    }
+
+    return lines.join('\n');
+  }
+
+  appTrace(): string {
+    if (this.appCallTrace === undefined || !this.disassembly) return '';
+    return this.trace(this.appCallTrace, this.disassembly);
+  }
+
+  lsigTrace(): string {
+    if (
+      this.logicSigTrace === undefined ||
+      this.logicSigDisassemly === undefined
+    )
+      return '';
+    return this.trace(this.logicSigTrace, this.logicSigDisassemly);
+  }
+}
+
+export class DryrunResult {
+  error: string = '';
+  protocolVersion: string = '';
+  txns: DryrunTransactionResult[] = [];
+  constructor(drrResp: Record<string, any>) {
```

Comment:
All the constructors take `Record<string, any>`, but they could be more specific. It seems like every constructor accesses a deterministic set of keys, so you could type these as object interfaces instead.

`Record<string, any>` is a step up from `any`, but not a big step 😅